src/emmy/calculus/derivative.cljc

```diff
@@ -202,18 +202,19 @@
 ;; preserved through tag replacement and extraction.
 
 (extend-protocol d/IPerturbed
-
   MultiFn
   (perturbed? [_] false)
   (replace-tag [f old new] (replace-tag-fn f old new))
-  (extract-tangent [f tag] (extract-tangent-fn f tag))
+  (extract-tangent [f tag mode] (extract-tangent-fn f tag mode))
+  (extract-id [f id] (comp #(d/extract-id % id) f))
```

Comment:
it's kind of an `fmap` for most types... after your multimethod conversion, @littleredcomputer , do you feel like these should be multimethods too? Then we could have a `::functor` type that supports fmap?